lib/runtime/wasmer/exports.go

```diff
@@ -227,6 +229,56 @@ func (in *Instance) QueryCallFeeDetails(ext []byte) (*types.FeeDetails, error) {
 // TODO: use this in block verification process (#1873)
 func (in *Instance) CheckInherents() {}
 
+// GrandpaGenerateKeyOwnershipProof returns grandpa key ownership proof from runtime.
+func (in *Instance) GrandpaGenerateKeyOwnershipProof(authSetID uint64, authorityID ed25519.PublicKeyBytes) (
+	types.OpaqueKeyOwnershipProof, error) {
+
+	const bufferSize = 8 + 32 // authSetID uint64 + ed25519.PublicKeyBytes
+	buffer := bytes.NewBuffer(make([]byte, 0, bufferSize))
+	encoder := scale.NewEncoder(buffer)
+	err := encoder.Encode(authSetID)
+	if err != nil {
+		return nil, fmt.Errorf("encoding auth set id: %w", err)
+	}
+	err = encoder.Encode(authorityID)
+	if err != nil {
+		return nil, fmt.Errorf("encoding authority id: %w", err)
+	}
+	encodedOpaqueKeyOwnershipProof, err := in.Exec(runtime.GrandpaGenerateKeyOwnershipProof, buffer.Bytes())
+	if err != nil {
+		return nil, err
+	}
+
+	keyOwnershipProof := types.OpaqueKeyOwnershipProof{}
+	err = scale.Unmarshal(encodedOpaqueKeyOwnershipProof, &keyOwnershipProof)
+	if err != nil {
+		return nil, fmt.Errorf("scale decoding: %w", err)
+	}
+
+	return keyOwnershipProof, nil
+}
+
+// GrandpaSubmitReportEquivocationUnsignedExtrinsic reports equivocation report to the runtime.
```

Comment:
supernit to avoid confusion with `reports` + `report` 😄 

```suggestion
// GrandpaSubmitReportEquivocationUnsignedExtrinsic reports an equivocation report to the runtime.
```